### PR TITLE
Combine ci linter jobs

### PR DESCRIPTION
**Pull request by tobiasdiez**
to de-clutter the "checks" a bit.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by kwankyu** [comment]
@mkoeppe ?

**Review by kwankyu — APPROVED** [review]
LGTM.

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -14,60 +14,32 @@ concurrency:
...
     - name: Set up Python
       uses: actions/setup-python@v4
       with:
         python-version: 3.9
-    - name: Install pycodestyle
-      run: pip install tox pycodestyle
-    - name: Lint using pycodestyle
+
+    - name: Install dependencies
+      run: pip install tox pycodestyle relint
```
Comment (mkoeppe) [inline .github/workflows/lint.yml:36]: you don't need to manually install pycodestyle, relint.

tox installs them in its virtual environments automaticallly.

**Comment by mkoeppe** [comment]
It no longer runs the other linters if the first one failed. That's a regression.
e.g. https://github.com/sagemath/sage/actions/runs/6599334105/job/17928241769

**Pull request merged**